### PR TITLE
Add XSDs for Format and Types.ps1xml files

### PR DESCRIPTION
**Pull request by felixfbecker**
## PR Summary

As proposed in #7749.

These can be used for autocompletion and validation in editors like VS Code. It makes writing Format and Types files ***so*** much more pleasant to write.

autocompletion

The schemas were taken from #7749 but extended with documentation parsed from the reference docs markdown [with a script](https://github.com/felixfbecker/PowerShellXSD).

Question: Should this be hosted on http://schemas.microsoft.com/ like the MAML schemas? How do the MAML schemas get published there?

cc @lzybkr @SteveL-MSFT

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: src/Schemas/Format.xsd
Comment (SteveL-MSFT) [inline src/Schemas/Format.xsd]: `.NET Framework` is specifically not `.NET Core`.  Since this applies to both, I would suggest just changing all occurrences of `.NET Framework` to simply `.NET`

File: src/Schemas/Format.xsd
Comment (SteveL-MSFT) [inline src/Schemas/Format.xsd]: Perhaps make this a bit more generic in case this get extended:

> Defines the properties of the column.  Examples: label, width, and alignment.

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]
Great contribution!  Didn't review the schema itself in detail, but overall it looks fine.  We can address as bugs if issues are reported.

**Comment by felixfbecker** [comment]
@SteveL-MSFT as mentioned, all documentation is 1:1 unmodified from the Format XML schema reference: https://docs.microsoft.com/en-us/powershell/developer/format/format-schema-xml-reference

I can easily update these schemas with new documentation by running the script again, but I don't think it makes sense to apply fixes to the documentation only to the schema? Most people will probably read the docs online. We should decide which one is the source of truth

**Comment by SteveL-MSFT** [comment]
@felixfbecker I submitted [https://github.com/PowerShell/PowerShell-Docs/pull/2943] to fix the online documentation.  I'm ok leaving the second comment until we add more to the schema.  You can wait for that to be merged and rerun your script.

**Comment by lzybkr** [comment]
I like the idea of a single source of truth. Maybe we should include your script in this PR so anyone can update the schemas from the docs.

**Comment by felixfbecker** [comment]
In an ideal world the docs would rather be generated from the schema (there are probably tools out there that do this). Parsing markdown with regexes is really hacky so I'm not sure if we want to rely on that. The current docs are not very well-structured in general so generating them differently would likely do them good.

@SteveL-MSFT your change actually didn't affect my script because it only updated the table of contents, I sent a follow-up PR: https://github.com/PowerShell/PowerShell-Docs/pull/2944

**Comment by felixfbecker** [comment]
My script also detected many cases in the docs where the schema you gave me @lzybkr 	directly contradicted the reference docs, e.g. an explicit `minOccurs="0"` in the schema, but an explicit "This element is required" in the docs. I tried out a few cases by omitting the element, loading the file into PowerShell and checking if it errors, and found that in all cases the docs were wrong. So the docs are a bad single source of truth.

**Comment by lzybkr** [comment]
Generating the markdown does sound easier. It doesn't need to be in this PR, but I do like that idea.

And I don't see every truly having a single source of truth without generating PowerShell's parser from the xsd - but having done that once before, we'd want the generated code to be much better than what I saw a few years ago.

**Comment by felixfbecker** [comment]
@lzybkr @SteveL-MSFT updated schemas with latest docs, should be good to merge now.

File: src/Schemas/Types.xsd
Comment (SteveL-MSFT) [inline src/Schemas/Types.xsd]: Please add a trailing newline at the end

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

**Comment by SteveL-MSFT** [comment]
@felixfbecker please rebase off master to pick up fixes causing the CI failures

**Comment by felixfbecker** [comment]
@SteveL-MSFT all green ✅

**Review by SteveL-MSFT — APPROVED** [review]

**Comment by felixfbecker** [comment]
@adityapatwardhan?

**Comment by adityapatwardhan** [comment]
@felixfbecker Sorry for the delay. I will review this today.

**Review by adityapatwardhan — APPROVED** [review]

**Comment by adityapatwardhan** [comment]
@felixfbecker Thanks for your contribution!

**Pull request merged**
